File: 3345/solution.java
class Solution {

    boolean isProductDivisible(int digit, int t){
        int curr = 1;
        while(digit>0){
            curr = curr * (digit%10);
            digit/=10;
        }
        return curr%t==0;
    }
    
    public int smallestNumber(int n, int t) {
        int res = n;
        while(!isProductDivisible(res,t))
            res++;
        return res;
    }
}

File: 3345/solution.py
class Solution:
    def smallestNumber(self, n: int, t: int) -> int:
        def isProductDivisible(digit, t):
            curr = 1 
            while digit:
                curr = curr*(digit%10)
                digit//=10 
            return curr%t 
        
        res = n 
        while isProductDivisible(res, t):
            res+=1 
        return res
